projects/addon-editor/components/editor/editor.component.ts

```diff
@@ -164,6 +165,15 @@ export class TuiEditorComponent
         this.editor?.unsetLink();
     }
 
+    focus(event: MouseEvent): void {
+        if (tuiIsNativeFocused(this.nativeFocusableElement)) {
```

Comment:
You only need to act if clicked target is not inside the editable part. Otherwise first focusing click sets caret at the beginning and not at the place inside the text that was clicked.

Reply:
fixed